### PR TITLE
feat: [FC-0047] Account Settings + Profile split

### PR DESCRIPTION
**Pull request by PavloNetrebchuk**
### Separation account settings from the user profile

https://www.figma.com/file/iZ56YMjbRMShCCDxqrqRrR/Open-edX-Mobile-App-All-Screens-v2.1?type=design&node-id=12831-300341&mode=design&t=3XfyM3YlwL4jSw1b-0

| Discover screen | Dashboard screen | Profile |
| ------------- | ------------- | ------------- |
|Screenshot 2024-04-10 at 15 44 13|Screenshot 2024-04-10 at 15 44 26|Screenshot 2024-04-10 at 15 44 40|

| Light theme | Dark theme |
| ------------- | ------------- |
|Screenshot 2024-04-10 at 16 55 11|Screenshot 2024-04-10 at 16 55 28|
|Screenshot 2024-04-10 at 15 35 46|Screenshot 2024-04-10 at 15 38 37|
|Screenshot 2024-04-10 at 15 39 12|Screenshot 2024-04-10 at 15 39 33|
|Screenshot 2024-04-10 at 15 40 07|Screenshot 2024-04-10 at 15 40 29|
|Screenshot 2024-04-10 at 15 40 47|Screenshot 2024-04-10 at 15 41 05|

| Landscape | Tablet |
| ------------- | ------------- |
|Screenshot 2024-04-10 at 15 47 53|Screenshot 2024-04-10 at 15 48 10|
|Screenshot 2024-04-10 at 15 48 38|Screenshot 2024-04-10 at 15 48 54|

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @PavloNetrebchuk! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

File: app/src/main/java/org/openedx/app/di/ScreenModule.kt [hunk 1]
```diff
@@ -152,6 +152,8 @@ val screenModule = module {
     viewModel { (qualityType: String) -> VideoQualityViewModel(qualityType, get(), get(), get()) }
     viewModel { DeleteProfileViewModel(get(), get(), get(), get(), get()) }
     viewModel { (username: String) -> AnothersProfileViewModel(get(), get(), username) }
+    viewModel{ SettingsViewModel(get(), get(), get(), get(), get(), get(), get(), get(), get()) }
```
Comment (volodymyr-chekyrta) [inline app/src/main/java/org/openedx/app/di/ScreenModule.kt:155]:
```suggestion
    viewModel { SettingsViewModel(get(), get(), get(), get(), get(), get(), get(), get(), get()) }
```

File: dashboard/src/main/java/org/openedx/dashboard/presentation/DashboardFragment.kt [hunk 2]
```diff
@@ -240,7 +244,32 @@ internal fun MyCoursesScreen(
...
+                    style = MaterialTheme.appTypography.titleMedium
+                )
+                IconButton(
+                    modifier = Modifier
+                        .align(Alignment.CenterEnd)
+                        .padding(end = 12.dp),
+                    onClick = { onSettingsClick() }
+                ) {
+                    Icon(
+                        painter = painterResource(id = org.openedx.core.R.drawable.core_ic_settings),
```
Comment (volodymyr-chekyrta) [inline dashboard/src/main/java/org/openedx/dashboard/presentation/DashboardFragment.kt:267]:
```suggestion
                        painter = painterResource(id = CoreR.drawable.core_ic_settings),
```

File: dashboard/src/main/java/org/openedx/dashboard/presentation/DashboardFragment.kt [hunk 3]
```diff
@@ -240,7 +244,32 @@ internal fun MyCoursesScreen(
...
+                        tint = MaterialTheme.appColors.primary,
+                        contentDescription = stringResource(id = org.openedx.core.R.string.core_accessibility_settings)
```
Comment (volodymyr-chekyrta) [inline dashboard/src/main/java/org/openedx/dashboard/presentation/DashboardFragment.kt:269]:
```suggestion
                        contentDescription = stringResource(id = CoreR.string.core_accessibility_settings)
```

File: discovery/src/main/java/org/openedx/discovery/presentation/NativeDiscoveryFragment.kt [hunk 4]
```diff
@@ -308,22 +318,41 @@ internal fun DiscoveryScreen(
...
+                        style = MaterialTheme.appTypography.titleMedium
+                    )
+                    IconButton(
+                        modifier = Modifier
+                            .align(Alignment.CenterEnd)
+                            .padding(end = 12.dp),
+                        onClick = { onSettingsClick() }
+                    ) {
+                        Icon(
+                            painter = painterResource(id = org.openedx.core.R.drawable.core_ic_settings),
```
Comment (volodymyr-chekyrta) [inline discovery/src/main/java/org/openedx/discovery/presentation/NativeDiscoveryFragment.kt:345]:
```suggestion
                            painter = painterResource(id = CoreR.drawable.core_ic_settings),
```

File: discovery/src/main/java/org/openedx/discovery/presentation/NativeDiscoveryFragment.kt [hunk 5]
```diff
@@ -308,22 +318,41 @@ internal fun DiscoveryScreen(
...
+                            tint = MaterialTheme.appColors.primary,
+                            contentDescription = stringResource(id = org.openedx.core.R.string.core_accessibility_settings)
```
Comment (volodymyr-chekyrta) [inline discovery/src/main/java/org/openedx/discovery/presentation/NativeDiscoveryFragment.kt:347]:
```suggestion
                            contentDescription = stringResource(id = CoreR.string.core_accessibility_settings)
```

File: profile/src/main/java/org/openedx/profile/presentation/delete/DeleteProfileFragment.kt [hunk 6]
```diff
@@ -193,103 +187,104 @@ fun DeleteProfileScreen(
...
-                    Spacer(Modifier.height(24.dp))
+                    Column(
+                        modifier = contentWidth
+                    ) {
+                        Spacer(Modifier.height(48.dp))
+                        Image(
+                            modifier = Modifier
+                                .size(145.dp)
+                                .align(Alignment.CenterHorizontally),
+                            painter = painterResource(id = org.openedx.profile.R.drawable.profile_delete_box),
```
Comment (volodymyr-chekyrta) [inline profile/src/main/java/org/openedx/profile/presentation/delete/DeleteProfileFragment.kt:220]:
```suggestion
                            painter = painterResource(id = profileR.drawable.profile_delete_box),
```

File: profile/src/main/java/org/openedx/profile/presentation/manage_account/ManageAccountViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    private val _uiMessage = MutableLiveData<UIMessage>()
+    val uiMessage: LiveData<UIMessage>
+        get() = _uiMessage
+
+    private val _isUpdating = MutableLiveData<Boolean>()
+    val isUpdating: LiveData<Boolean>
+        get() = _isUpdating
+    init {
+        getAccount()
+    }
```
Comment (volodymyr-chekyrta) [inline profile/src/main/java/org/openedx/profile/presentation/manage_account/ManageAccountViewModel.kt:57]:
```suggestion
    private val _isUpdating = MutableLiveData<Boolean>()
    val isUpdating: LiveData<Boolean>
        get() = _isUpdating
        
    init {
        getAccount()
    }
```

File: profile/src/main/java/org/openedx/profile/presentation/manage_account/compose/ManageAccountView.kt [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+                                    Spacer(modifier = Modifier.height(12.dp))
+                                    ProfileTopic(
+                                        image = if (uiState.account.profileImage.hasImage) uiState.account.profileImage.imageUrlFull else null,
+                                        title = uiState.account.name,
+                                        subtitle = uiState.account.email ?: ""
+                                    )
+                                    OpenEdXOutlinedButton(
+                                        modifier = Modifier
+                                            .fillMaxWidth(),
+                                        text = stringResource(id = org.openedx.profile.R.string.profile_edit_profile),
```
Comment (volodymyr-chekyrta) [inline profile/src/main/java/org/openedx/profile/presentation/manage_account/compose/ManageAccountView.kt:174]:
```suggestion
                                        text = stringResource(id = profileR.string.profile_edit_profile),
```

File: profile/src/main/java/org/openedx/profile/presentation/manage_account/compose/ManageAccountView.kt [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+                                        onClick = {
+                                            onAction(ManageAccountViewAction.EditAccountClick)
+                                        },
+                                        borderColor = MaterialTheme.appColors.buttonBackground,
+                                        textColor = MaterialTheme.appColors.textAccent
+                                    )
+                                    Spacer(modifier = Modifier.height(12.dp))
+                                    IconText(
+                                        text = stringResource(id = org.openedx.profile.R.string.profile_delete_profile),
+                                        painter = painterResource(id = org.openedx.profile.R.drawable.profile_ic_trash),
```
Comment (volodymyr-chekyrta) [inline profile/src/main/java/org/openedx/profile/presentation/manage_account/compose/ManageAccountView.kt:184]:
```suggestion
                                        text = stringResource(id = profileR.string.profile_delete_profile),
                                        painter = painterResource(id = profileR.drawable.profile_ic_trash),
```

File: profile/src/main/java/org/openedx/profile/presentation/manage_account/compose/ManageAccountView.kt [hunk 10]
```diff
@@ -0,0 +1,267 @@
...
+            windowSize = WindowSize(WindowType.Medium, WindowType.Medium),
+            uiState = mockUiState,
+            uiMessage = null,
+            refreshing = false,
+            onAction = {}
+        )
+    }
+}
+
+val mockAccount = Account(
```
Comment (volodymyr-chekyrta) [inline profile/src/main/java/org/openedx/profile/presentation/manage_account/compose/ManageAccountView.kt:239]: Can we make it private?

File: profile/src/main/java/org/openedx/profile/presentation/settings/SettingsFragment.kt [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+
+                            SettingsScreenAction.ManageAccount -> {
+                                viewModel.manageAccountClicked(
+                                    requireActivity().supportFragmentManager
+                                )
+                            }
+                        }
+                    }
+                )
+
```
Comment (volodymyr-chekyrta) [inline profile/src/main/java/org/openedx/profile/presentation/settings/SettingsFragment.kt:95]:
```suggestion
```

File: profile/src/main/java/org/openedx/profile/presentation/settings/SettingsScreenUI.kt [hunk 12]
```diff
@@ -0,0 +1,702 @@
...
+            }
+        }
+    }
+}
+
+@Composable
+private fun ManageAccountSection(onManageAccountClick: () -> Unit) {
+    Column {
+        Card(
+            modifier = Modifier,
```
Comment (volodymyr-chekyrta) [inline profile/src/main/java/org/openedx/profile/presentation/settings/SettingsScreenUI.kt:237]: Unused modifier

File: profile/src/main/java/org/openedx/profile/presentation/settings/SettingsScreenUI.kt [hunk 13]
```diff
@@ -0,0 +1,702 @@
...
+)
+
+private val mockUiState = SettingsUIState.Data(
+    configuration = mockConfiguration
+)
+
+
+@Preview
+@Composable
+fun AppVersionItemAppToDatePreview() {
```
Comment (volodymyr-chekyrta) [inline profile/src/main/java/org/openedx/profile/presentation/settings/SettingsScreenUI.kt:650]:
```suggestion
private fun AppVersionItemAppToDatePreview() {
```

File: profile/src/main/java/org/openedx/profile/presentation/settings/SettingsScreenUI.kt [hunk 14]
```diff
@@ -0,0 +1,702 @@
...
+            versionName = mockAppData.versionName,
+            appUpgradeEvent = null,
+            onClick = {}
+        )
+    }
+}
+
+@Preview
+@Composable
+fun AppVersionItemUpgradeRecommendedPreview() {
```
Comment (volodymyr-chekyrta) [inline profile/src/main/java/org/openedx/profile/presentation/settings/SettingsScreenUI.kt:662]:
```suggestion
private fun AppVersionItemUpgradeRecommendedPreview() {
```

File: profile/src/main/java/org/openedx/profile/presentation/settings/SettingsScreenUI.kt [hunk 15]
```diff
@@ -0,0 +1,702 @@
...
+            versionName = mockAppData.versionName,
+            appUpgradeEvent = AppUpgradeEvent.UpgradeRecommendedEvent("1.0.1"),
+            onClick = {}
+        )
+    }
+}
+
+@Preview
+@Composable
+fun AppVersionItemUpgradeRequiredPreview() {
```
Comment (volodymyr-chekyrta) [inline profile/src/main/java/org/openedx/profile/presentation/settings/SettingsScreenUI.kt:674]:
```suggestion
private fun AppVersionItemUpgradeRequiredPreview() {
```

File: profile/src/main/java/org/openedx/profile/presentation/settings/SettingsScreenUI.kt [hunk 16]
```diff
@@ -0,0 +1,702 @@
...
+            versionName = mockAppData.versionName,
+            appUpgradeEvent = AppUpgradeEvent.UpgradeRequiredEvent,
+            onClick = {}
+        )
+    }
+}
+
+@Preview
+@Composable
+fun LogoutDialogPreview() {
```
Comment (volodymyr-chekyrta) [inline profile/src/main/java/org/openedx/profile/presentation/settings/SettingsScreenUI.kt:686]:
```suggestion
private fun LogoutDialogPreview() {
```

File: profile/src/main/java/org/openedx/profile/presentation/ui/ProfileUI.kt [hunk 17]
```diff
@@ -107,50 +98,19 @@ fun ProfileInfoSection(account: Account) {
...
-                                    style = SpanStyle(
-                                        color = MaterialTheme.appColors.textPrimaryVariant
-                                    ),
-                                    start = 0,
-                                    end = text.length - value.length
-                                )
-                            },
-                            style = MaterialTheme.appTypography.titleMedium,
+                            modifier = Modifier.fillMaxWidth(),
+                            text = stringResource(id = org.openedx.profile.R.string.profile_about_me),
```
Comment (volodymyr-chekyrta) [inline profile/src/main/java/org/openedx/profile/presentation/ui/ProfileUI.kt:106]:
```suggestion
                            text = stringResource(id = profileR.string.profile_about_me),
```

File: profile/src/main/java/org/openedx/profile/presentation/ui/SettingsUI.kt [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+            style = MaterialTheme.appTypography.titleMedium,
+            color = MaterialTheme.appColors.textPrimary
+        )
+        Icon(
+            modifier = Modifier.size(16.dp),
+            imageVector = icon,
+            contentDescription = null
+        )
+    }
+}
```
Comment (volodymyr-chekyrta) [inline profile/src/main/java/org/openedx/profile/presentation/ui/SettingsUI.kt:61]:
```suggestion
}

```

File: profile/src/main/java/org/openedx/profile/presentation/video/VideoSettingsFragment.kt [hunk 19]
```diff
@@ -0,0 +1,317 @@
...
+                                .height(92.dp)
+                                .clickable {
+                                    videoStreamingQualityClick()
+                                },
+                            horizontalArrangement = Arrangement.SpaceBetween,
+                            verticalAlignment = Alignment.CenterVertically
+                        ) {
+                            Column(Modifier.weight(1f)) {
+                                Text(
+                                    text = stringResource(id = org.openedx.core.R.string.core_video_streaming_quality),
```
Comment (volodymyr-chekyrta) [inline profile/src/main/java/org/openedx/profile/presentation/video/VideoSettingsFragment.kt:246]:
```suggestion
                                    text = stringResource(id = CoreR.string.core_video_streaming_quality),
```

File: profile/src/main/java/org/openedx/profile/presentation/video/VideoSettingsFragment.kt [hunk 20]
```diff
@@ -0,0 +1,317 @@
...
+                                .height(92.dp)
+                                .clickable {
+                                    videoDownloadQualityClick()
+                                },
+                            horizontalArrangement = Arrangement.SpaceBetween,
+                            verticalAlignment = Alignment.CenterVertically
+                        ) {
+                            Column(Modifier.weight(1f)) {
+                                Text(
+                                    text = stringResource(id = org.openedx.core.R.string.core_video_download_quality),
```
Comment (volodymyr-chekyrta) [inline profile/src/main/java/org/openedx/profile/presentation/video/VideoSettingsFragment.kt:277]:
```suggestion
                                    text = stringResource(id = CoreR.string.core_video_download_quality),
```

File: profile/src/main/java/org/openedx/profile/presentation/video/VideoSettingsFragment.kt [hunk 21]
```diff
@@ -0,0 +1,317 @@
...
+                                Text(
+                                    text = stringResource(id = videoSettings.videoDownloadQuality.titleResId),
+                                    color = MaterialTheme.appColors.textSecondary,
+                                    style = MaterialTheme.appTypography.labelMedium
+                                )
+                            }
+                            Icon(
+                                imageVector = Icons.Filled.ChevronRight,
+                                tint = MaterialTheme.appColors.onSurface,
+                                contentDescription = "Expandable Arrow"
```
Comment (volodymyr-chekyrta) [inline profile/src/main/java/org/openedx/profile/presentation/video/VideoSettingsFragment.kt:291]: Same here

**Review by volodymyr-chekyrta — CHANGES_REQUESTED** [review]

File: profile/src/main/java/org/openedx/profile/presentation/manage_account/compose/ManageAccountView.kt (hunk 10 above)
Comment (PavloNetrebchuk) [reply profile/src/main/java/org/openedx/profile/presentation/manage_account/compose/ManageAccountView.kt:239]: We use this code in multiple files, so I relocated it to the ProfileUi

File: profile/src/main/java/org/openedx/profile/presentation/manage_account/compose/ManageAccountView.kt [hunk 22]
```diff
@@ -0,0 +1,248 @@
+package org.openedx.profile.presentation.manage_account.compose
```
Comment (volodymyr-chekyrta) [inline profile/src/main/java/org/openedx/profile/presentation/manage_account/compose/ManageAccountView.kt:1]: No underscores for package names
https://google.github.io/styleguide/javaguide.html#s5.2.1-package-names
`manage_account` -> `manage`
`manage_account` -> `manageaccount`
or other names

**Review by volodymyr-chekyrta — APPROVED** [review]

**Comment by volodymyr-chekyrta** [comment]
Ready for product review

File: core/src/main/java/org/openedx/core/ui/ComposeExtensions.kt
Comment (k1rill) [inline core/src/main/java/org/openedx/core/ui/ComposeExtensions.kt:1]: new line at the EOF

**Review by k1rill — DISMISSED** [review]

**Review by HamzaIsrar12 — CHANGES_REQUESTED** [review]
Awesome work overall. 🏎️ 

Only some nits from the refactoring:
- The settings icon is missing from `WebViewDiscoveryFragment` and `ProgramFragment`; both based on webviews
- We have the same `logout` function in three different ViewModels. Please only retain the one in `SettingsViewModel` and only listen to `AccountDeactivated` event in the `SettingsViewModel`.
- Do we need to add the Delete Account button in Edit Profile? 🤔 If not, then we can delete some code around logout notifiers and success events.

File: profile/src/main/java/org/openedx/profile/presentation/manageaccount/compose/ManageAccountView.kt [hunk 23]
```diff
@@ -0,0 +1,248 @@
...
+                )
+            )
+        }
+
+        HandleUIMessage(uiMessage = uiMessage, scaffoldState = scaffoldState)
+
+        Column(
+            modifier = Modifier
+                .padding(paddingValues)
+                .settingsHeaderBackground()
```
[the remark this answers is not in the document]
Comment (PavloNetrebchuk) [reply profile/src/main/java/org/openedx/profile/presentation/manageaccount/compose/ManageAccountView.kt:112]: I was also confused when the background disappeared from the design.😁 
That's ok. As I know, the background will change soon. Am I right? @volodymyr-chekyrta

File: profile/src/main/java/org/openedx/profile/presentation/manageaccount/compose/ManageAccountView.kt [hunk 24]
```diff
@@ -0,0 +1,248 @@
...
+                                Box(
+                                    modifier = Modifier.fillMaxSize(),
+                                    contentAlignment = Alignment.Center
+                                ) {
+                                    CircularProgressIndicator(color = MaterialTheme.appColors.primary)
+                                }
+                            }
+
+                            is ManageAccountUIState.Data -> {
+                                Column(
```
[the remark this answers is not in the document]
Comment (PavloNetrebchuk) [reply profile/src/main/java/org/openedx/profile/presentation/manageaccount/compose/ManageAccountView.kt:157]: The addition of those buttons will be included in the upcoming pull requests.

**Review by HamzaIsrar12 — CHANGES_REQUESTED** [review]
(From the previous review)
Do we need to add the Delete Account button in Edit Profile? 🤔 If not, then we can delete some code around logout notifiers and success events.

File: profile/src/main/java/org/openedx/profile/presentation/ui/SettingsUI.kt [hunk 25]
```diff
@@ -0,0 +1,61 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import org.openedx.core.extension.tagId
+import org.openedx.core.ui.theme.appColors
+import org.openedx.core.ui.theme.appTypography
+
+@Composable
+fun SettingsItem(
```
[the remark this answers is not in the document]
Comment (PavloNetrebchuk) [reply profile/src/main/java/org/openedx/profile/presentation/ui/SettingsUI.kt:26]: Yes, for example, on the Manage Account screen. We currently do not have the "Change Email" and "Change Password" buttons, but they will be added later.
Screenshot 2024-04-25 at 15 22 43

File: profile/src/main/java/org/openedx/profile/presentation/video/VideoSettingsFragment.kt
[the remark this answers is not in the document]
Comment (PavloNetrebchuk) [reply profile/src/main/java/org/openedx/profile/presentation/video/VideoSettingsFragment.kt:1]: Okay, I'll give it a try. Thanks for the tip!

**Comment by PavloNetrebchuk** [comment]
@HamzaIsrar12 @k1rill 
Done✅ Could you please check and confirm if everything is correct?

**Review by HamzaIsrar12 — APPROVED** [review]
I  just completed the final pass and everything looks good to me. 🏎️ ✨ 

_But please also get the approval from Volodymyr on this [thread](https://github.com/openedx/openedx-app-android/pull/286#discussion_r1578203669)._

**Comment by volodymyr-chekyrta** [comment]
> I just completed the final pass and everything looks good to me. 🏎️ ✨
> 
> _But please also get the approval from Volodymyr on this [thread](https://github.com/openedx/openedx-app-android/pull/286#discussion_r1578203669)._

@HamzaIsrar12 thank you!
I've double-checked this [thread](https://github.com/openedx/openedx-app-android/pull/286#discussion_r1578203669) 👍

**Review by volodymyr-chekyrta — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@PavloNetrebchuk 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-app-android&entry.867055334=https://github.com/openedx/openedx-app-android/pull/286&entry.1484655318=https://github.com/PavloNetrebchuk&entry.752974735=2024-04-08+16:20&entry.1917517419=2024-04-29+19:56&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**